packages/blade/src/components/Switch/__tests__/Switch.web.test.tsx

```diff
@@ -0,0 +1,169 @@
+    const user = userEvents.setup();
+    const name = 'Toggle darkmode';
+    const { getByRole } = renderWithTheme(<Switch accessibilityLabel={name} />);
+
+    expect(getByRole('switch', { name })).not.toBeChecked();
+    await user.tab();
+    expect(getByRole('switch', { name })).toHaveFocus();
+    await user.keyboard('[Space]');
+    expect(getByRole('switch', { name })).toBeChecked();
+    await user.keyboard('[Space]');
+    expect(getByRole('switch', { name })).not.toBeChecked();
+  });
+
+  it('should set defaultChecked', () => {
+    const name = 'Toggle Darkmode';
+    const { getByRole } = renderWithTheme(<Switch defaultChecked accessibilityLabel={name} />);
+
+    expect(getByRole('switch', { name })).toBeChecked();
+  });
+
+  it('should support isChecked prop', async () => {
+    const user = userEvents.setup();
+    const name = 'Toggle Darkmode';
+    const { getByRole } = renderWithTheme(<Switch isChecked accessibilityLabel={name} />);
+
+    expect(getByRole('switch', { name })).toBeChecked();
+    // should not toggle
+    await user.click(getByRole('switch', { name }));
+    expect(getByRole('switch', { name })).toBeChecked();
+  });
+
+  it('should support uncontrolled state', async () => {
+    const user = userEvents.setup();
+    const checkFn = jest.fn();
+    const name = 'Toggle Darkmode';
+    const { getByRole } = renderWithTheme(
+      <Switch
+        value="darkmode"
+        name="darkmode-toggle"
+        accessibilityLabel={name}
+        defaultChecked={true}
+        onChange={checkFn}
+      />,
+    );
+
+    expect(getByRole('switch', { name })).toBeChecked();
+    expect(checkFn).not.toBeCalled();
+    await user.click(getByRole('switch', { name }));
+    expect(getByRole('switch', { name })).not.toBeChecked();
+    expect(checkFn).toBeCalledWith(
+      expect.objectContaining({
+        isChecked: false,
+        value: 'darkmode',
+      }),
+    );
+    expect(checkFn.mock.calls[0][0].event).not.toBeUndefined();
+    await user.click(getByRole('switch', { name }));
+    expect(getByRole('switch', { name })).toBeChecked();
+    expect(checkFn).toBeCalledWith(
+      expect.objectContaining({
+        isChecked: true,
+        value: 'darkmode',
+      }),
+    );
+    expect(checkFn.mock.calls[0][0].event).not.toBeUndefined();
+  });
+
+  it('should support controlled state', async () => {
+    const user = userEvents.setup();
+    const name = 'Toggle Darkmode';
+    const Example = () => {
+      const [checked, setChecked] = React.useState(false);
+      return (
+        <>
+          <Switch
+            accessibilityLabel={name}
+            isChecked={checked}
+            onChange={({ isChecked }) => setChecked(isChecked)}
+          />
+          <p data-testid="state">{checked ? 'checked' : 'unchecked'}</p>
+        </>
+      );
+    };
+    const { getByRole, getByTestId } = renderWithTheme(<Example />);
+
+    expect(getByTestId('state')).toHaveTextContent('unchecked');
```

Comment:
Instead of adding a test id you can test on the text content directly, expect text content to be there